manager/dispatcher/nodes.go

```diff
@@ -86,8 +89,13 @@ func (s *nodeStore) CheckRateLimit(id string) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	if existRn, ok := s.nodes[id]; ok {
+		existRn.Attempts++
 		if time.Since(existRn.Registered) < s.rateLimitPeriod {
-			return grpc.Errorf(codes.Unavailable, "node %s attempted registration too recently", id)
+			if existRn.Attempts > rateLimitCount {
+				return grpc.Errorf(codes.Unavailable, "node %s attempted registration too recently", id)
+			}
+		} else {
+			existRn.Attempts = 0
```

Comment:
1 feels more logical to me because we're counting attempts, and the client just made an attempt. But it doesn't matter except for changing how `rateLimitCount` is interpreted.